### PR TITLE
Use inheritance for debug labels and add them to descriptors

### PR DESCRIPTION
**Pull request by Kangz**
PTAL, not sure if this is good practice in WebIDL or not.

### CODE REVIEW

File: design/sketch.webidl [hunk 1]
```diff
@@ -21,6 +21,12 @@ dictionary GPUExtent3D {
     u32 depth;
 };
 
+interface GPUObjectBase {
+};
+
+dictionnary GPUObjectDescriptorBase {
```
[the remark this answers is not in the document]
Comment (Kangz) [reply design/sketch.webidl]: Whoops, that's a frenchism :) Done

**Comment by grorg** [comment]
Discussed at the [1 April Teleconference](https://docs.google.com/document/d/1k_sueTXngp4XtXLyVTBJ9p-5zt3N6xcQ4yP0rDVS0ng/edit)

**Comment by kainino0x** [comment]
To merge after moving the 2 partials up into the original definitions, and after #248.

**Comment by kainino0x** [comment]
ping: #248 has landed

**Comment by Kangz** [comment]
Updated to index.bs and folded partial interfaces.

**Pull request merged**